sdk/samples/compute/armcompute/main.go

```diff
@@ -0,0 +1,347 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/armcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/sdk/to"
+)
+
+var (
+	subscriptionId    string
+	location          = "westus2"
+	resourceGroupName = "sample-resourcegroup"
+	vnetName          = "sample-vnet"
+	subnetName        = "internal"
+	nicName           = "sample-nic"
+	vmName            = "sample-vm"
+)
+
+func init() {
+	subscriptionId = os.Getenv("AZURE_SUBSCRIPTION_ID")
+}
+
+func main() {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
+	}
+	conn := armcore.NewDefaultConnection(cred, &armcore.ConnectionOptions{
+		Logging: azcore.LogOptions{
+			IncludeBody: true,
+		},
+	})
+
+	ctx := context.Background()
+	resourceGroup, err := createResourceGroup(ctx, conn)
+	if err != nil {
+		log.Fatalf("Cannot create resource group: %+v", err)
+	}
+	log.Printf("Resource Group ID: %s", *resourceGroup.ID)
+
+	vnet, err := createVirtualNetwork(ctx, conn)
+	if err != nil {
+		log.Fatalf("Cannot create virtual network: %+v", err)
+	}
+	log.Printf("Virtual Network ID: %s", *vnet.ID)
+
+	subnet, err := createSubnet(ctx, conn)
+	if err != nil {
+		log.Fatalf("Cannot create subnet: %+v", err)
+	}
+	log.Printf("Subnet ID: %s", *subnet.ID)
+
+	nic, err := createNIC(ctx, conn, *subnet.ID)
+	if err != nil {
+		log.Fatalf("Cannot create network interface: %+v", err)
+	}
+	log.Printf("Network Interface ID: %s", *nic.ID)
+
+	vm, err := createVirtualMachine(ctx, conn, *nic.ID)
+	if err != nil {
+		log.Fatalf("Cannot create virtual machine: %+v", err)
+	}
+	log.Printf("Virtual Machine ID: %s", *vm.ID)
+}
+
+func createResourceGroup(ctx context.Context, connection *armcore.Connection) (*armresources.ResourceGroup, error) {
+	rgClient := armresources.NewResourceGroupsClient(connection, subscriptionId)
+
+	param := armresources.ResourceGroup{
+		Location: to.StringPtr(location),
+	}
+
+	resp, err := rgClient.CreateOrUpdate(ctx, resourceGroupName, param, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.ResourceGroup, nil
+}
+
+func updateResourceGroup(ctx context.Context, connection *armcore.Connection) (*armresources.ResourceGroup, error) {
```

Comment:
Seems in main(), only create methods are used, is it possible to also reference update/list/delete method in the main function?